cli/cmd/install.go

```diff
@@ -516,9 +516,23 @@ func (options *installOptions) recordableFlagSet() *pflag.FlagSet {
 		"Enables installing the SMI-Metrics controller",
 	)
 
-	flags.StringVarP(&options.controlPlaneVersion, "control-plane-version", "", options.controlPlaneVersion, "(Development) Tag to be used for the control plane component images")
+	flags.StringVarP(&options.controlPlaneVersion, "control-plane-version", "", options.controlPlaneVersion, "Tag to be used for the control plane component images")
 	flags.StringVar(&options.smiMetricsImage, "smi-metrics-image", options.smiMetricsImage, "SMI Metrics image")
-	flags.MarkHidden("control-plane-version")
+
+	// Hide developer focused flags in release builds.
+	release, err := version.IsReleaseChannel(version.Version)
+	if err != nil {
+		log.Errorf("Unable to parse version: %s", version.Version)
+	}
+	if release {
+		flags.MarkHidden("control-plane-version")
+		flags.MarkHidden("proxy-image")
+		flags.MarkHidden("proxy-version")
```

Comment:
FWIW, I think I'd be comfortable leaving `proxy-version` exposed. We ask users to use it frequently to test fixes... But I can ask them to use hidden flags, too...